docs/source/upgrade.rst

```diff
@@ -362,6 +363,8 @@ For performance reasons, :func:`cupy.count_nonzero` has been changed to return z
 See the discussion in `#154 <https://github.com/cupy/cupy/pull/154>`_ for more details.
 
 
+.. _compatibility_matrix:
```

Comment:
```
:ref:`compatibility matrix`
```
(case-insensitive) would link to the section without the annotation.

https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#sections

> Each section title automatically generates a hyperlink target pointing to the section. The text of the hyperlink target (the "reference name") is the same as that of the section title. See [Implicit Hyperlink Targets](https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#implicit-hyperlink-targets) for a complete description.